src/NHibernate.Test/Linq/MiscellaneousTextFixture.cs

```diff
@@ -86,16 +86,17 @@ public void ReferenceToOuter()
 							"the second, third and fourth pages of products")]
 		public void TriplePageSelection()
 		{
-			IQueryable<Product> q = (
-			                        	from p in db.Products
-			                        	where p.ProductId > 1
-			                        	orderby p.ProductId
-			                        	select p
-			);
-
-            IQueryable<Product> page2 = q.Skip(5).Take(5);
-            IQueryable<Product> page3 = q.Skip(10).Take(5);
-            IQueryable<Product> page4 = q.Skip(15).Take(5);
+			var q =
+				from p in db.Products
+				where p.ProductId > 1
+				orderby p.ProductId
+				select p;
+
+			// ToList required otherwise the First call alters the paging and test something else than paging three pages.
```

Comment:
`Skip(5).Take(5).First()` does not make sens, it should be `Skip(5).First()`. Do you really think the test writer intended the previous code?

For me, this test is testing something else that what it was intended for. Something that bugs with Oracle by the way. But that bug is a bit corner case, the `Take` has no purpose anymore when calling `First` on it.